src/calliope/attrdict.py

```diff
@@ -359,13 +355,14 @@ def to_yaml(self, path=None):
         # handle multi-line strings.
         walk_tree(result)
 
-        if path is not None:
+        stream = io.StringIO()
+        yaml_.dump(result, stream)
+        yaml_str = stream.getvalue()
+        if path:
+            yaml_str = self.to_yaml()
             with open(path, "w") as f:
-                yaml_.dump(result, f)
-        else:
-            stream = io.StringIO()
-            yaml_.dump(result, stream)
-            return stream.getvalue()
+                f.write(yaml_str)
```

Comment:
why do you need `self.to_yaml()` here? you can use the constructed `yaml_str` directly. You can also use pathlib to avoid the context manager: `Path(path).write_text(yaml_str)`

Reply:
Whoops. That's what you get from copy pasting!